splunk-otel-android/src/main/java/com/splunk/rum/DiskToZipkinExporter.java

```diff
@@ -0,0 +1,182 @@
+        this.bandwidthTracker = builder.bandwidthTracker;
+        this.bandwidthLimit = builder.bandwidthLimit;
+    }
+
+    void startPolling() {
+        threadPool.scheduleAtFixedRate(this::doExportCycle, 5, 5, TimeUnit.SECONDS);
+    }
+
+    //Visible for testing
+    void doExportCycle() {
+        try {
+            exportPendingFiles();
+        } catch (Exception e) {
+            Log.w(LOG_TAG, "Error processing on-disk files", e);
+        }
+    }
+
+    private void exportPendingFiles() throws IOException {
+        if (!connectionUtil.refreshNetworkStatus().isOnline()) {
+            Log.i(SplunkRum.LOG_TAG, "Network offline, leaving spans on disk for for eventual export.");
+            return;
+        }
+
+        List<Path> pendingFiles = getPendingFiles();
+        boolean sentAnything = false;
+        for (Path file : pendingFiles) {
+
+            double sustainedRate = bandwidthTracker.totalSustainedRate();
+            if (sustainedRate > bandwidthLimit) {
+                Log.i(SplunkRum.LOG_TAG, String.format("Export rate %.2f exceeds limit of %.2f, backing off", sustainedRate, bandwidthLimit));
+                break;
+            }
+
+            boolean dataWasSent = handleFileOnDisk(file);
+            sentAnything |= dataWasSent;
+            if (!dataWasSent) {   // Don't bother trying any remaining files if this one failed.
+                break;
+            }
+        }
+        if (!sentAnything) {
+            bandwidthTracker.tick(emptyList());
+        }
+    }
+
+    private List<Path> getPendingFiles() throws IOException {
+        return fileUtils.listFiles(spanFilesPath)
+                .filter(fileUtils::isRegularFile)
+                .filter(file -> file.toString().endsWith(".spans"))
+                .sorted(Comparator.comparing(o -> o.getFileName().toString()))
+                .collect(Collectors.toList());
+    }
+
+    private boolean handleFileOnDisk(Path file) {
+        Log.d(LOG_TAG, "Reading file content for ingest: " + file);
+        List<byte[]> encodedSpans = readFileCompletely(file);
+        if(encodedSpans.isEmpty()) {
+            return false;
+        }
+
+        try {
+            bandwidthTracker.tick(encodedSpans);
+            Call<Void> httpCall = sender.sendSpans(encodedSpans);
+            httpCall.execute();
+            Log.d(LOG_TAG, "File content " + file + " successfully uploaded");
+            return true;
+        } catch (IOException e) {
+            Log.w(LOG_TAG, "Error sending file content", e);
+            return false;
+        }
+        finally {
+            fileUtils.safeDelete(file);
+        }
+    }
+
+    private List<byte[]> readFileCompletely(Path file) {
+        try {
+            return fileUtils.readFileCompletely(file);
+        } catch (IOException e) {
+            Log.w(LOG_TAG, "Error reading span data from file " + file, e);
+            return emptyList();
+        }
+    }
+
+    void stop() {
```

Comment:
Is this method called anywhere?

Reply:
It's not. Let me circle back to see where in the lifecycle it _should_ be called...

Reply:
Well, since the lifecycle of the exporter matches the lifecycle of `SplunkRum` and it doesn't look like that gets shut down...probably given this comment 
<img width="666" alt="image" src="https://user-images.githubusercontent.com/75337021/158225734-eb27f863-e28f-4346-9215-6d38a033eb84.png">

Do you think we should just remove the `stop()` method here then?

Reply:
Hm, let's remove it - if we ever need to support stopping the whole `SplunkRum` then we'll probably have to do a lot more work than just that one method.